src/main/java/core/basesyntax/Main.java

```
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.ReportExporterDao;
+import core.basesyntax.dao.ReportExporterDaoImpl;
+import core.basesyntax.dao.ReportExtractorDao;
+import core.basesyntax.dao.ReportExtractorDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.ReportIsNullException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ProductCounterService;
+import core.basesyntax.service.ReportCreator;
+import core.basesyntax.service.ReportDataParserService;
+import core.basesyntax.service.impl.ProductCounterServiceImpl;
+import core.basesyntax.service.impl.ReportCreatorImpl;
+import core.basesyntax.service.impl.ReportDataParserServiceImpl;
+import java.io.File;
+import java.util.List;
+import static core.basesyntax.strategy.ActionConfig.createResolver;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class Main {
+    private static final String DATA_FILE = "report.csv";
+    private static final String REPORT_FILE = "src/main/resources/finalReport.csv";
+    // HINT: In the `public static void main(String[] args)` it is better to create instances of your classes,
+    // and call their methods, but do not write any business logic in the `main` method!
+    public static void main(String[] args) {
+        //Зчитування файлу
+        Storage storage = new Storage();
+        ClassLoader classLoader = Main.class.getClassLoader();
+        File file;
+        try {
+            file = new File(classLoader.getResource(DATA_FILE).getFile());
+        } catch (NullPointerException e) {
+            throw new ReportIsNullException("Resource " + DATA_FILE + " not found on classpath");
```

This block catches NullPointerException when trying to load the resource. Instead of catching NPE, perform an explicit null-check of classLoader.getResource(DATA_FILE) and throw an informative RuntimeException if it's null. This addresses the task requirement to "Handle incorrect file path (throw informative RuntimeException)" from the assignment description and avoids using exception-based control flow. See Main.java where getResource(...) is accessed and then NPE is caught. Reference: Main.java code uses classLoader.getResource(DATA_FILE).getFile() and catches NullPointerException. Also see checklist guidance: "Don't ignore exceptions in try-catch blocks. Don't leave `e.printStackTrace()` or `System.out.println(e.getMessage())` in the catch blocks. Let's rethrow a RuntimeException with an **informative** message and exception object."